src/MongoDB.Driver.Core/Core/Configuration/ClusterBuilderExtensions.cs

```diff
@@ -60,8 +60,21 @@ public static ClusterBuilder ConfigureWithConnectionString(this ClusterBuilder b
 
             if (!connectionString.IsResolved)
             {
-                var connectionMode = connectionString.Connect;
-                var resolveHosts = connectionMode == ClusterConnectionMode.Direct || connectionMode == ClusterConnectionMode.Standalone;
+                bool resolveHosts;
+#pragma warning disable CS0618
+                if (connectionString.ConnectionModeSwitch == ConnectionModeSwitch.UseDirectConnection)
+#pragma warning restore CS0618
+                {
+                    resolveHosts = connectionString.DirectConnection.GetValueOrDefault();
+                }
+                else
+                {
+#pragma warning disable CS0618
+                    var connectionMode = connectionString.Connect;
+                    resolveHosts = connectionMode == ClusterConnectionMode.Direct || connectionMode == ClusterConnectionMode.Standalone;
```

Comment:
Replace with:  

    resolveHosts = connectionString.Connect == ClusterConnectionMode.Direct || connectionString.Connect == ClusterConnectionMode.Standalone;

Reply:
Or rename `connectionMode` temporary variable to avoid clash with suggested change below.